### DISCUSSION TITLE
How to apply it in NextJS

### DISCUSSION THREAD

**Discussion by bigomise** [eclipse-langium/langium#1199] [Q&A]
Hi,

I'm interested in Langium but i'm struggling in implementing with NextJS app. Not sure how to use the wrapper and also worker in NextJS.

If anyone could share an example would be appreciated 🙏.

Thanks in advance!

**Comment by msujew** [comment]
Hey @bigomise,

I assume you're using React with NextJS? The monaco wrapper only requires an HTML element to mount into. You should be able to use a `useRef` with a `useEffect` and mount the editor into the underlying HTML element. You can also use the [`@typefox/monaco-editor-react`](https://www.npmjs.com/package/@typefox/monaco-editor-react) component, which is a ready to use React component for the wrapper.

As for the worker, you should be able to just move/compile the language server into your `public` folder and reference the webworker as a path. In theory you should also be able to just use the path to the TypeScript file that is the entry point to your language server. See [an example here](https://github.com/TypeFox/monaco-components/blob/4e08fc3047808e8f3a8d380578d5abb04ce8fda4/packages/examples/src/langium/wrapperLangium.ts#L69-L77) (compiled with vite). And another one with [NextJS](https://stackblitz.com/github/vercel/next.js/tree/canary/examples/with-web-worker) (but without Langium).

**Reply by bigomise** [reply]
wow, thanks a lot.. let me take a look at it. @msujew  🙏
